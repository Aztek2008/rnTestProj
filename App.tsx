import React from 'react';
import {Provider} from 'react-redux';
import NavBar from './components/NavBar';
import {store} from './redux/ConfigureStore';
import {NavigationContainer} from '@react-navigation/native';
// import {StatusBar, useColorScheme} from 'react-native';

// import {Colors} from 'react-native/Libraries/NewAppScreen';

const App = () => {
  // const isDarkMode = useColorScheme() === 'dark';

  // const backgroundStyle = {
  //   backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  // };
  // style={backgroundStyle}>

  return (
    <Provider store={store}>
      <NavigationContainer>
        {/* <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} /> */}
        <NavBar />
      </NavigationContainer>
    </Provider>
  );
};

export default App;
