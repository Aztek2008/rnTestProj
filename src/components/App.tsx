import {Provider} from 'react-redux';
import React, {useEffect} from 'react';
import NavBar from './NavBar';
import {store} from '../redux/ConfigureStore';
import messaging from '@react-native-firebase/messaging';
import {NavigationContainer} from '@react-navigation/native';
import {Alert, StatusBar, Text, useColorScheme, View} from 'react-native';
import {requestUserPermission, unsubscribe} from '../notification/FCMService';
// import firebase from '@react-native-firebase';
// import {Colors} from 'react-native/Libraries/NewAppScreen';
// type Notification = {body: string; title: string} | undefined;

const App = () => {
  // const isDarkMode = useColorScheme() === 'dark';
  // const backgroundStyle = {
  //   backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  // };
  // style={backgroundStyle}>

  messaging().setBackgroundMessageHandler(async remoteMessage => {
    console.log('Message handled in the background!', remoteMessage);
  });

  useEffect(() => {
    requestUserPermission();

    return unsubscribe;
  }, []);

  return (
    <Provider store={store}>
      <NavigationContainer>
        {/* <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} /> */}
        <NavBar />
      </NavigationContainer>
    </Provider>
  );
};

export default App;
