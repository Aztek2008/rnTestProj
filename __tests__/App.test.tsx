// /**
//  * @format
//  */
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/components/App';

// Avoiding ReferenceError: You are trying to `import` a file after the Jest environment has been torn down.
jest.useFakeTimers();

jest.mock(
  'react-native-vector-icons/FontAwesome',
  () => 'MockedFontAwesomeIcons',
);

test('App renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
