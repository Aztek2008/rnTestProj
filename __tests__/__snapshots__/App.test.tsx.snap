// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="#88C1CB"
        color="#fff"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={false}
        largeTitleHideShadow={false}
        title="Main"
        titleColor="#fff"
        titleFontWeight="bold"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": "50%",
                "position": "relative",
                "width": "50%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "position": "absolute",
                }
              }
            >
              <MockedFontAwesomeIcons
                color="#323232"
                name="edit"
                size={50}
              />
              <Text>
                Add User
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": "50%",
                "position": "relative",
                "width": "50%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "position": "absolute",
                }
              }
            >
              <MockedFontAwesomeIcons
                color="#323232"
                name="users"
                size={50}
              />
              <Text>
                All Users
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": "50%",
                "position": "relative",
                "width": "50%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "position": "absolute",
                }
              }
            >
              <MockedFontAwesomeIcons
                color="#323232"
                name="info"
                size={50}
              />
              <Text>
                Device Info
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": "50%",
                "position": "relative",
                "width": "50%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "position": "absolute",
                }
              }
            >
              <MockedFontAwesomeIcons
                color="#323232"
                name="camera"
                size={50}
              />
              <Text>
                Camera
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": "50%",
                "position": "relative",
                "width": "50%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "position": "absolute",
                }
              }
            >
              <MockedFontAwesomeIcons
                color="#323232"
                name="photo"
                size={50}
              />
              <Text>
                Pictures
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
                "opacity": 1,
                "paddingTop": "50%",
                "position": "relative",
                "width": "50%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "position": "absolute",
                }
              }
            >
              <MockedFontAwesomeIcons
                color="#323232"
                name="photo"
                size={50}
              />
              <Text>
                Crypto
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
